apm-agent-tracer/src/main/java/co/elastic/apm/agent/tracer/Tracer.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.apm.agent.tracer;
+
+import co.elastic.apm.agent.tracer.dispatch.BinaryHeaderGetter;
+import co.elastic.apm.agent.tracer.dispatch.HeaderGetter;
+import co.elastic.apm.agent.tracer.dispatch.TextHeaderGetter;
+import co.elastic.apm.agent.tracer.pooling.ObjectPoolFactory;
+
+import javax.annotation.Nullable;
+
+public interface Tracer {
+
+    boolean isRunning();
+
+    @Nullable
+    <T extends Tracer> T probe(Class<T> type);
+
+    <T extends Tracer> T require(Class<T> type);
```

Comment:
[question] I'm guessing the use-case for those methods might come from next steps in this refactoring, but do we need to have this kind of cast directly available in the `Tracer` API ?

Reply:
This could be added in a next step, but I wanted to add it as part of the minimal API. The idea is to allow for casting the internal tracer of a global tracer to a specific implementation without the global tracer needing to know about a specific implementation.

This way it will still be possible to receive the `ElasticApmTracer`, or any other specific tracer implementation from a global tracer.